e2e/core/BUILD.bazel

```diff
@@ -34,37 +40,23 @@ write_file(
 #     tools = [":node_bin"],
 # )
 
-# In theory, you can use the node toolchain together with a genrule().
-# However the genrule implementation doesn't perform toolchain resolution.
-# See this comment from Jay Conrod about a similar question for rules_go
-# https://github.com/bazelbuild/rules_go/issues/2255#issuecomment-545478712
-# That means you must resolve the toolchain yourself, with a select()
-# and that falls down the same deoptimization described above:
-# it will eager-fetch node for all platforms.
-# So instead we recommend always writing a custom rule to access the node binary.
-# alias(
-#     name = "node_toolchain",
-#     actual = select({
-#         "@bazel_tools//src/conditions:darwin_arm64": "@node16_darwin_arm64//:node_toolchain",
-#         "@bazel_tools//src/conditions:darwin_x86_64": "@node16_darwin_amd64//:node_toolchain",
-#         "@bazel_tools//src/conditions:linux_aarch64": "@node16_linux_arm64//:node_toolchain",
-#         "@bazel_tools//src/conditions:linux_s390x": "@node16_linux_s390x//:node_toolchain",
-#         "@bazel_tools//src/conditions:linux_x86_64": "@node16_linux_amd64//:node_toolchain",
-#         "@bazel_tools//src/conditions:linux_ppc64le": "@node16_linux_ppc64le//:node_toolchain",
-#         "@bazel_tools//src/conditions:windows": "@node16_windows_amd64//:node_toolchain",
-#         "//conditions:default": "@node16_linux_amd64//:node_toolchain",
-#     }),
-# )
-# genrule(
-#     name = "use_node_toolchain",
-#     srcs = ["some.js"],
-#     outs = ["thing2"],
-#     cmd = "$(NODE_PATH) $(execpath some.js) $@",
-#     toolchains = [":node_toolchain"],
-#     # It will also fail to include the files from the node_toolchain, so you're
-#     # forced to repeat the label of the node binary as an explicit input.
-#     tools = ["@node16_host//:node_bin"],
-# )
+# You can use the node toolchain together with a genrule().
+# This gives you complete control over starting the interpreter, but you also have to
+# manually handle module resolution.
+genrule(
+    name = "use_node_toolchain",
+    srcs = ["some.js"],
+    outs = ["genrule_out"],
+    cmd = "$(NODE_PATH) $(execpath some.js) $@",
+    toolchains = ["@node16_toolchains//:resolved_toolchain"],
+    tools = ["@node16_toolchains//:resolved_toolchain"],
```

Comment:
Do you actually need both of these? That's kind of ugly. Do you think runfiles provided via `toolchains` should also be available like `tools` are? Or is this clear enough?

Reply:
I ran into exactly the same ugliness in https://github.com/bazelbuild/rules_perl/pull/27 :) It would be nice to implicitly add to runfiles, but not a huge deal

Reply:
Can one of you file an issue about this, so I don't lose track?

Reply:
https://github.com/bazelbuild/bazel/issues/14014